test/worker_test.py

```diff
@@ -305,6 +305,27 @@ def run(self, tracking_url_callback=None):
         # Should only run and fail once, not retry because of the type error
         self.assertEqual(1, a.num_runs)
 
+    def test_tracking_url(self):
+        tracking_url = 'http://test_url.com/'
+
+        class A(Task):
+            has_run = False
+
+            def complete(self):
+                return self.has_run
+
+            def run(self):
+                if self.set_tracking_url is not None:
```

Comment:
Wait. Do the user really have to do this check before using the tracking_url? Can't we somehow guarantee that it's always present?